### ISSUE TITLE
[3.4.0] iOS AudioEngine.play(clip) cause memory leak on iOS build.

### ISSUE THREAD

**Issue by thinhlv** [cocos/cocos-engine#11508]
### Cocos Creator version

3.4.0

### System information

Iphone 8, iOS 15.4

### Issue description

- AudioSource play audio background music cause memory leak for every time we changes the music clip.
- Create a simple script to change your  background music by time, on the native platform (iOS 15.4).
- The memory leak will be accured for every music changes.
- Look at the Instrument result, the cc::AudioEngine::play2d() cause the leak.

### Steps to reproduce

- Add the blow script to your scene,
- Create audio Source and assign to the component
- Import 3 audio clip for your music background, and assign to the component
- Build project to iOS native
- Open xcode Allocation Instrument, watch the leaks result.

Result: Memory leak every 10 seconds, when the background music has been changed.
![Screen Shot 2022-06-15 at 09 54 31](https://user-images.githubusercontent.com/18746281/173727821-794d2e4a-0e8c-44e0-91fa-a62eca9ed098.png)
 

### Minimal reproduction project

```typescript
import { _decorator, Component, Node, AudioSource, AudioClip } from 'cc';
const { ccclass, property } = _decorator;

@ccclass('MusicTest')
export class MusicTest extends Component {
    @property(AudioSource)
    source: AudioSource = null;

    @property([AudioClip])
    clips: Array<AudioClip> = [];               //3 items.

    idx: number = 0;
    count: number = 0;

    update(deltaTime) {
        this.count -= deltaTime;
        if (this.count <= 0) {
            this.count = 10;

            // this.source.stop();
            // this.source.clip = null;

            this.source.clip = this.clips[this.idx];
            this.source.play();

            this.idx++;
            if (this.idx >= 3) {
                this.idx = 0;
            }
        }
    }
}
```

**Comment by minggo** [comment]
@thinhlv thanks for feedback. Can i know more detail informations:
- Does it only cause leak on iOS 15.4? Have you tested on other platforms?
- Have you tested other iOS devices?

**Comment by minggo** [comment]
As this issue is reported later, do not have to fix it before 6.20.

**Comment by thinhlv** [comment]
> 

I'm not sure for other devices and os versions. I have just tested on only my phone and ios 15.4. Thanks

**Comment by pandamicro** [comment]
This will be delayed to v3.6.1, @timlyeee please keep track and give feedbacks here

**Comment by timlyeee** [comment]
![image](https://user-images.githubusercontent.com/43355299/189873098-0caf7009-9f71-4f02-bb16-b654fcab1c1e.png)

The reason for the memory leak here is caused by the unsafe free.
As you can see on v3.4.2, it use `free(element)` to free the memory while element is actually constructed by `new`.

Replace the code with `delete(element)` resolve this issue. The bug is also fixed on v3.6
